### DISCUSSION TITLE
Meeting Notes Automation

### DISCUSSION THREAD

**Discussion by SlinkyPotato** [degen-workgroup/DEGEN#104] [Ideas]
Taken from Discord:

Let me try and break it down how something that like might be doable (very rough draft)
1. auto creation of templatized agenda in Notion
 - /create notes guild: 'dev guild'
 - generates page with the template for dev guild in the notion site under dev guild notes
2. auto sharing agenda
 - access discord voice api and look under dev-workroom 
 - send a message to all those attending
3. auto bounty posted ~30m before meeting for notetaker
 - bounty gets posted by user who initiated /create notes
 - uses existing bounty flow
4. maybe signoff on notes? (hard to evaluate what "good quality" means)
 - at end of discord voice session send a message to executor of command -/create notes
 - send url link of notes location
 - reply with emoji if notes look good
 - if it looks good, then bounty is closed and creator can either tip or let bounty distribute funds from escrow
 - if not then bounty creator should go on bountyboard site when ready or reach out to support

**Comment by lvnacy** [comment]
How do you think the logic will work for this? Will each guild have a separate module that will create a meeting page on Notion in their respective guild pages? Because if it's similar to the welcome mats, I'm wondering if it will make sense to start setting up mini "packages" for each guild, so when creating a command that will have granular access to guilds, we just create a supporting module for each guild and place it in their respective "package." So, the folder structure could be such:
```
app
|__ commands
      |__ meetingPrepAndBounty.ts
|__ errors
|__ events
      |__ welcomeMats.ts
|__ guilds
      |__ index.ts (collect all the default exports into one object and export the object)
      |__ devGuild
            |__ welcomeMat.ts
            |__ meetingPrep.ts
      |__ writersGuild
            |__ welcomeMat.ts
            |__ meetingPrep.ts
```

This way, we don't have multiple places where separate guild modules are stored. And we can simply chain access to methods:
```
sendWelcomeMatMessage.devGuild.devWelcome(blah: more blah);
```

Thoughts?

**Reply by SlinkyPotato** [reply]
This looks good! The way I see a guild is that it represents an entity and to me an entity is like a data model so we could place the guilds folder under /repository/guilds or even /models/guilds. Either works well and supports future work for other data operations that might be needed. 

I'd imagine the guilds would each have their own requirements or things they'd like to have as part of the automation flow. I can see each guild folder keeping track of that and looking it up in notion (i.e additional columns in the notion table for guild A vs guild B)

To start I'd say we would create a notion table for each guild and in that table a page is generated that would be used for notes. 

Let's only do this for one guild first:
1. To see how it works once it's out
2. If other guilds want this, they can put in a bounty for it
3. It can even be like a demo kind of thing during community call, everyone will see it, love it and put in bounties

**Reply by lvnacy** [reply]
Excellent. Yeah, we can name the folders whatever. Just wanted to start modularizing the support files for individual guilds to keep everything clean and easily accessible. Great feedback.
